### ISSUE TITLE
OTP-24: possible bypass of after block in try .. catch .. after ... end in case of error in catch block

### ISSUE THREAD

**Issue by kpy3** [erlang/otp#4859]
**Describe the bug**
Erlang VM bypasses `after` block completely then catch block contains `erlang:raise/3` call only (see example below). Adding something BEFORE `erlang:raise/3` (`erlang:display/1` in example below) eliminates this bug, and everything seem works as expected.

**To Reproduce**
```
-module(bug).
-export([test/0]).

test() ->
    Pid = erlang:spawn(
      fun() ->
              receive
                  {From, Msg} ->
                      try
                          call(Msg)
                      catch
                          E:R:S ->
                              %erlang:display('!'),
                              erlang:raise(E,R,S)
                      after
                          erlang:display('+'),
                          From ! Msg
                      end
              end
      end),
    Pid ! {self(), test},
    receive
        test -> no_bug
    after
        1000 -> bug_found
    end.

call(_Msg) ->
    error(undefined).
```

With commented out `erlang:display/1` it gives:
```
❯ erl
Erlang/OTP 24 [erts-12.0] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit] [dtrace]

Eshell V12.0  (abort with ^G)
1> c(bug).
{ok,bug}
2> bug:test().
=ERROR REPORT==== 20-May-2021::18:55:52.877152 ===
Error in process <0.90.0> with exit value:
{undefined,[{bug,call,1,[{file,"bug.erl"},{line,29}]},
            {bug,'-test/0-fun-0-',0,[{file,"bug.erl"},{line,10}]}]}

bug_found
```

And with `erlang:display/1` in catch:
```
❯ erl
Erlang/OTP 24 [erts-12.0] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit] [dtrace]

Eshell V12.0  (abort with ^G)
1> c(bug).
{ok,bug}
2> bug:test().
'!'
'+'
no_bug
=ERROR REPORT==== 20-May-2021::18:57:13.965662 ===
Error in process <0.90.0> with exit value:
{undefined,[{bug,call,1,[{file,"bug.erl"},{line,29}]},
            {bug,'-test/0-fun-0-',0,[{file,"bug.erl"},{line,10}]}]}
```

**Expected behavior**
Executing `after` block should not depend on contents of `catch` block.

**Affected versions**
Affects Erlang 24.0.0

**Comment by jhogberg** [comment]
Thanks for your report! I've fixed it in https://github.com/erlang/otp/pull/4864.

**Comment by jhogberg** [comment]
We've released a fix in OTP-24.0.2, thanks again for your report!
